src/raw/client.rs

```diff
@@ -818,6 +800,58 @@ impl<PdC: PdClient> Client<PdC> {
         Ok(result)
     }
 
+    async fn retryable_scan(
+        &self,
+        mut scan_args: ScanInnerArgs,
+    ) -> Result<(Option<RawScanResponse>, Key)> {
+        let start_key = scan_args.start_key;
+
+        let region = self.rpc.clone().region_for_key(&start_key).await?;
+        let store = self.rpc.clone().store_for_id(region.id()).await?;
+        let request = new_raw_scan_request(
+            scan_args.range.clone(),
+            scan_args.limit,
+            scan_args.key_only,
+            scan_args.reverse,
+            self.cf.clone(),
+        );
```

Comment:
@pingyu  The problem might be over here, we are not retrieving the new regions or stores once it is updated. This may need to be moved into the loop.